File: ng/src/app/models/expense-summary-dto.ts
import { Category } from '@/models/category';

export class ExpenseSummaryDto {
    totalAmount?: number;
    categoryId?: number;
    expenseCount?: number;
    category?: Category;
    budgetUsedPercentage?: number;
    remainingBudget?: number;
    budgetAmount?: number;
}

File: api/mathew/dtos/ExpenseSummaryDto.cs
public class ExpenseSummaryDto
{
    public int CategoryId { get; set; }
    public decimal TotalAmount { get; set; }
    public int ExpenseCount { get; set; }
    public decimal BudgetAmount { get; set; }
    public decimal RemainingBudget { get; set; }
    public decimal BudgetUsedPercentage { get; set; }
}